pos_customer_required/static/src/js/Screens/PartnerListScreen/PartnerListScreen.esm.js

```diff
@@ -0,0 +1,27 @@
+/** @odoo-module **/
+
+/* Copyright CoopITEasy - Simon Hick <sim@coopiteasy.be>
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl) */
+
+import PartnerListScreen from "point_of_sale.PartnerListScreen";
+import Registries from "point_of_sale.Registries";
+
+const PosCustomerRequiredPartnerListScreen = (OriginalPartnerListScreen) =>
+    class extends OriginalPartnerListScreen {
+        clickPartner(partner) {
+            if( this.env.pos.config.require_customer === "order" ) {
+                if (this.state.selectedPartner && this.state.selectedPartner.id === partner.id) {
+                    this.state.selectedPartner = null;
+                    this.render(true);
```

Comment:
this should not be needed, should it?

Reply:
`this.render(true);` was needed to reload the screen to show the user had been unselected. But now that the logic is moved to confirm() `this.state.selectedPartner = null;` will be handled in clickPartner already, so I've removed it.